boilerplates/trpc/files/trpc/server.ts

```diff
@@ -29,6 +46,7 @@ export const appRouter = router({
       todoItems.push({ text: opts.input });
       return { todoItems };
     }),
```

Comment:
`onCreateTodo` and `onNewTodo` should be merged into a unique function. Wrap the mutation content in `if (BATI.has(...)) { ... }` block to achieve the same thing.

Reply:
I've tried that way before.
```ts
if (BATI.has("drizzle")) {
  const result: RunResult = await db.insert(todoTable).values({ text: opts.input });
  return { result };
} else {
  todoItems.push({ text: opts.input });
  return { todoItems };
}
```
But, I think it makes typescript confused, so we cannot do object destructuring.

Reply:
we only need that object destructuring for vue, because `await reload()` doesn't work for vue

Reply:
Not sure to understand, what object destructuring are you talking about here?
Also, why do we need to call `await reload()` at all?

Reply:
This one https://github.com/batijs/bati/pull/258/files#diff-c3e53e4a3a9deeb2ebe0a3607a624c98feb9c25a03ccc4a72e2263ff2e12d0eaR33

We need to call `await reload()` so that the todo list is updated

Reply:
Neat, thanks for your answer.
If I may, I think I have another approach, let see if you disagree then I'll try yours. :pray:

Reply:
Yeah sure

Reply:
Please take a look at this commit [ad8882d](https://github.com/batijs/bati/pull/258/commits/ad8882d1d02fc7b31ea7af9918f023f57c803a5b).
I think, it's better to return something other than `true` or `false`. Users can than use that return to do setState like this :
```ts
setTodoItems((prev) => [...prev, { id: result.lastInsertRowid as number, text: newTodo }]);
```

In this way, the page is not reloaded, and the data is not refetched. But still updated